src/solidity-frontend/solidity_convert.cpp

```diff
@@ -1566,6 +1566,22 @@ bool solidity_convertert::get_noncontract_defition(nlohmann::json &ast_node)
     // for abstract contract
     add_empty_body_node(ast_node);
   }
+  else if (
+    node_type == "ContractDefinition" && ast_node["contractKind"] == "library")
+  {
+    std::string lib_name = ast_node["name"].get<std::string>();
+
+    if (get_struct_class(ast_node))
+      return true;
+    current_baseContractName = lib_name;
```

Comment:
Please avoid overwriting the current_baseContractName before recording it first. 

Maybe something like:

old = current_baseContractName 
current_baseContractName  = xxx;

...

current_baseContractName  = old;